src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/AppManifestSpecialFileProvider.cs

```diff
@@ -47,14 +47,14 @@ protected override async Task CreateFileCoreAsync(string path)
         {
             ConfigurationGeneralBrowseObject configurationGeneral = await _properties.GetConfigurationGeneralBrowseObjectPropertiesAsync();
 
-            string value = (string)await configurationGeneral.ApplicationManifest.GetValueAsync();
-            if (value.Length == 0)
+            string? value = (string?)await configurationGeneral.ApplicationManifest.GetValueAsync();
+            if (string.IsNullOrEmpty(value))
```

Comment:
Behaviour change. Avoids NRE.

Reply:
These are annoying, when will this actually return null?

Reply:
Agree. I am not 100% certain when it can be null, but I know there are code paths that return null in CPS. I'm pretty sure if this API was designed with NRT in mind then a case like this would be handled differently.